tests/lammps/test_interactive.py

```diff
@@ -140,6 +140,23 @@ def test_calc_minimize_input(self):
         self.assertTrue(("fix ensemble all box/relax x 10000.0 y 20000.0 xy 0.0 xz 0.0 couple none" in
                          self.minimize_job._interactive_library._command))
 
+    def test_callback(self):
+        def f(x, nt, nl):
+            return np.linspace(0, 1, np.prod(x.shape)).reshape(-1, 3)
+        v = f(self.job.structure.positions, None, None)
+        v -= np.mean(v, axis=0)
+        self.job.server.run_mode.modal = True
+        self.assertRaises(AssertionError, self.job.set_callback, f)
+        self.job.server.run_mode.interactive = True
```

Comment:
```suggestion
        self.job.interactive_open()
```

And for good measure this should be coupled with a close at the end of the function, although I guess this class is inheriting from `TestWithCleanProject` such that `self.job` is anyhow getting safely destroyed. Still seems like best practice though!

Reply:
I guess there'd be a problem if the job is closed because it's also used elsewhere. Isn't it destroyed only when all tests are over?

Reply:
> I guess there'd be a problem if the job is closed because it's also used elsewhere. 

Aha. First, you're right -- it doesn't inherit from `TestWithCleanProject` at all, it's entirely written by hand. Using the same job elsewhere is super sketchy to me as it means behaviour in one test has a chance to infect behaviour in an independent test. There is some sort of cleaning happening in the [`setUp`](https://github.com/pyiron/pyiron_atomistics/blob/d2664693698f189e23a4fe3fbad603c4246d2c91/tests/lammps/test_interactive.py#L25) method, but I am not familiar enough with interactive jobs to say whether this is bullet-proof.

>  Isn't it destroyed only when all tests are over?

Yes, ultimately it gets killed in [`tearDownClass`](https://github.com/pyiron/pyiron_atomistics/blob/d2664693698f189e23a4fe3fbad603c4246d2c91/tests/lammps/test_interactive.py#L62)

I don't think fixing these things is the responsibility of this PR, but the current implementation of `TestLammpsInteractive` where the same job is reused in each test seems dangerous to me and should be fixed later in a separate PR.